warehouse/packaging/services.py

```diff
@@ -161,3 +167,70 @@ def remove_by_prefix(self, prefix):
             self.s3_client.delete_objects(
                 Bucket=self.bucket_name, Delete={"Objects": keys_to_delete}
             )
+
+
+@implementer(IFileStorage)
+class GCSFileStorage(GenericFileStorage):
+    @classmethod
+    def create_service(cls, context, request):
+        storage_client = request.find_service(name="gcloud.gcs")
+        bucket_name = request.registry.settings["files.bucket"]
+        bucket = storage_client.get_bucket(bucket_name)
+        prefix = request.registry.settings.get("files.prefix")
+
+        return cls(bucket, prefix=prefix)
+
+    def get(self, path):
+        # Note: this is not actually used in production, instead our CDN is
+        # configured to connect directly to our storage bucket. See:
+        # https://github.com/python/pypi-infra/blob/master/terraform/file-hosting/vcl/main.vcl
+        raise NotImplementedError
```

Comment:
Maybe it'd be worth adding a message similar to the note above? That way if it is raised we'd know why in the traceback without looking at the code

Reply:
Eh, I'm not so sure it's worth it. This is the only `NotImplementedError` in the entire codebase, and it's pretty unlikely it will get raised.

Reply:
Fair enough :)